client/src/main/java/org/eclipse/hono/client/impl/AbstractSender.java

```diff
@@ -314,9 +315,15 @@ protected final Span startSpan(final Message message) {
                     ServiceInvocationException e = null;
                     if (Rejected.class.isInstance(remoteState)) {
                         final Rejected rejected = (Rejected) remoteState;
-                        e = rejected.getError() == null
-                                ? new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST)
-                                : new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST, rejected.getError().getDescription());
+                        e = Optional.ofNullable(rejected.getError())
+                                .map(error -> {
+                                    if (AmqpError.RESOURCE_LIMIT_EXCEEDED.equals(error.getCondition())) {
+                                        return new ServerErrorException(HttpURLConnection.HTTP_UNAVAILABLE, error.getDescription());
+                                    } else {
+                                        return new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST, error.getDescription());
```

Comment:
There actually already is a method mapping an errorCondition to a `ServiceInvocationException`:

https://github.com/eclipse/hono/blob/e0ca00cc8b2925c79e91f35cd07a3ae217beeb39/client/src/main/java/org/eclipse/hono/client/StatusCodeMapper.java#L122-L127

It is only used in integration tests and its original intention seems to have been, to map errors created when opening an AMQP link (i.e. the failed result of a link openHandler).
Using the FORBIDDEN code seems appropriate in such a scenario, in contrast to the scenario addressed in this PR.

Therefore I don't want to suggest using that method here. Instead, to avoid confusion, I think the `StatusCodeMapper` method could be renamed to something like `fromConnectError`.

Reply:
Nice catch :+1: 
I will change the method name according to your suggestion ...

Reply:
@calohmn I have pushed a corresponding change ...

Reply:
Shouldn't this then also use `StatusCodeMapper#fromTransferError`?

Reply:
yes, it should